tests/test_http_client.py

```diff
@@ -52,6 +52,107 @@ def test_new_default_http_client_urllib2(self):
 
 
+class TestRetrySleepTimeDefaultHttpClient(StripeClientTestCase):
+    from contextlib import contextmanager
+
+    def assert_sleep_times(self, client, expected):
+        until = len(expected)
+        actual = list(map(lambda i: client._sleep_time(i+1), range(until)))
+        assert expected == actual
+
+    @contextmanager
+    def mock_max_delay(self, new_value):
+        original_value = stripe.http_client.HTTPClient.MAX_DELAY
+        stripe.http_client.HTTPClient.MAX_DELAY = new_value
+        try:
+            yield self
+        finally:
+            stripe.http_client.HTTPClient.MAX_DELAY = original_value
+
+    def test_sleep_time_exponential_back_off(self):
+        client = stripe.http_client.new_default_http_client()
+        client._add_jitter_time = lambda t: t
+        with self.mock_max_delay(10):
+            self.assert_sleep_times(client, [0.5, 1.0, 2.0, 4.0, 8.0])
+
+    def test_initial_delay_as_minimum(self):
+        client = stripe.http_client.new_default_http_client()
+        client._add_jitter_time = lambda t: t * 0.001
+        initial_delay = stripe.http_client.HTTPClient.INITIAL_DELAY
+        self.assert_sleep_times(client, [initial_delay] * 5)
+
+    def test_maximum_delay(self):
+        client = stripe.http_client.new_default_http_client()
+        client._add_jitter_time = lambda t: t
+        max_delay = stripe.http_client.HTTPClient.MAX_DELAY
+        expected = [0.5, 1.0, max_delay, max_delay, max_delay]
+        self.assert_sleep_times(client, expected)
+
+    def test_randomness_added(self):
+        client = stripe.http_client.new_default_http_client()
+        random_value = 0.8
+        client._add_jitter_time = lambda t: t * random_value
+        base_value = stripe.http_client.HTTPClient.INITIAL_DELAY * random_value
+
+        with self.mock_max_delay(10):
+            expected = [stripe.http_client.HTTPClient.INITIAL_DELAY,
+                        base_value * 2,
+                        base_value * 4,
+                        base_value * 8,
+                        base_value * 16]
+            self.assert_sleep_times(client, expected)
+
+    def test_jitter_has_randomness_but_within_range(self):
+        client = stripe.http_client.new_default_http_client()
+
+        jittered_ones = set(
+            map(lambda _: client._add_jitter_time(1), list(range(100)))
+        )
+
+        assert len(jittered_ones) > 1
+        assert all(0.5 <= val <= 1 for val in jittered_ones)
+
+
+class TestRetryConditionsDefaultHttpClient(StripeClientTestCase):
+
+    def test_should_retry_on_codes(self):
+        one_xx = list(range(100, 104))
+        two_xx = list(range(200, 209))
+        three_xx = list(range(300, 308))
+        four_xx = list(range(400, 431))
+        five_xx = list(range(500, 512))
+
+        client = stripe.http_client.new_default_http_client()
+        codes = one_xx + two_xx + three_xx + four_xx + five_xx
+        codes.remove(409)
```

Comment:
Haha, we probably don't need to be _this_ exhaustive on the tests, but definitely applaud your attention to detail ;)

(No action required — it's fine given it's done already.)